### ISSUE TITLE
One way road left side sharrows not shown or selectable, road is oneway also for cyclists (Right hand drive country)

### ISSUE THREAD

**Issue by hdevine825** [streetcomplete/StreetComplete#4701]
The left and right throughout are relevant to right hand drive countries, the inverse is likely true for left hand drive countries but I have not tested it. The term sharrows refers to the option in StreetComplete "lane shared explicitly with other traffic" and the OSM tag shared_lane.

Example way: https://www.openstreetmap.org/way/5681678
Aerial Image: 
![image](https://user-images.githubusercontent.com/12535864/209416281-c6fea3c6-c780-418a-b249-b1a5918a409a.png)

Issues:
1. The cycleway overlay currently shows a road with cycleway:left=shared_lane as having no cycleways mapped.
2. When selected to map cycleways the left side sharrow is not shown.
3. There is no way to select the sharrow for the left side. Including when selecting a different type of lane, reversing the direction, then attempting to select again.

**How to Reproduce**
Applies to one way roads with cycleway:left=shared_lane in the cycleway overlay.

**Expected Behavior**
1. The overlay should show an yellow/orange line on the left side of the way.
2. When the way is opened, sharrows should be shown on the left side going in the direction of car traffic.
3. It should be possible to select sharrows for the left side of the road going in the car traffic direction.

**Versions affected**
StreetComplete: v50.0
Android: 13

**Comment by matkoniecz** [comment]
important part of info: way is oneway also for cyclists and there is no contraflow here.

**Comment by westnordost** [comment]
> 1. The cycleway overlay currently shows a road with cycleway:left=shared_lane as having no cycleways mapped.
> 2. When selected to map cycleways the left side sharrow is not shown.

This is actually not a bug because it was not defined what kind of shared cycle lane it was.

When `cycleway:left:lane=pictogram` is supplied, StreetComplete does actually interpret the left cycleway to go into contraflow direction because if no `cycleway:left:oneway=*` is tagged, it assumes that the left side always is the contraflow side for right-hand-side-traffic countries (in relation to in which direction the road was drawn).
I think this is also correct. An explicit `cycleway:left:oneway=yes` is missing for https://www.openstreetmap.org/way/5681678

**Comment by westnordost** [comment]
However, I found the following issues:
The UI logic in the forms would not take into account a reversed cycleway (i.e. a cycleway that is specified to go into a non-default direction, such as a cycleway on the left side going forward). Among potentially other things, this meant:

- after reversing the direction to the non-default, the direction would reset after selecting another cycleway
- after reversing the direction to the non-default in a oneway, "none, but cyclists may use road in both directions" is still available as selection which does not make sense.
-after reversing the direction to the non-default in a oneway, "lane shared explicitly with other traffic" is not available as selection

This I fixed. So, while "lane shared explicitly with other traffic" is not available for selection directly, you will now be able to do the following:
1. select any other cycle lane
2. reverse it
3. now select the shared lane
